state-sync/aptos-data-client/src/priority.rs

```diff
@@ -0,0 +1,487 @@
+) -> PeerPriority {
+    // Handle the case that this node is a validator
+    let peer_network_id = peer.network_id();
+    if base_config.role.is_validator() {
+        // Validators should highly prioritize other validators
+        if peer_network_id.is_validator_network() {
+            return PeerPriority::HighPriority;
+        }
+
+        // VFNs should be prioritized over PFNs. Note: having PFNs
+        // connected to a validator is a rare (but possible) scenario.
+        return if peer_network_id.is_vfn_network() {
+            PeerPriority::MediumPriority
+        } else {
+            PeerPriority::LowPriority
+        };
+    }
+
+    // Handle the case that this node is a VFN
+    if peers_and_metadata
+        .get_registered_networks()
+        .contains(&NetworkId::Vfn)
+    {
+        // VFNs should highly prioritize validators
+        if peer_network_id.is_vfn_network() {
+            return PeerPriority::HighPriority;
+        }
+
+        // Trusted peers should be prioritized over untrusted peers.
+        // This prioritizes other VFNs/seed peers over regular PFNs.
+        if is_trusted_peer(peers_and_metadata.clone(), peer) {
+            return PeerPriority::MediumPriority;
+        }
+
+        // Outbound connections should be prioritized over inbound connections.
+        // This prioritizes other VFNs/seed peers over regular PFNs.
+        return if let Some(metadata) = utils::get_metadata_for_peer(&peers_and_metadata, *peer) {
+            if metadata.get_connection_metadata().is_outbound_connection() {
+                PeerPriority::MediumPriority
+            } else {
+                PeerPriority::LowPriority
+            }
+        } else {
+            PeerPriority::LowPriority // We don't have connection metadata
+        };
+    }
+
+    // Otherwise, this node is a PFN. PFNs should highly
+    // prioritize trusted peers (i.e., VFNs and seed peers).
+    if is_trusted_peer(peers_and_metadata.clone(), peer) {
+        return PeerPriority::HighPriority;
+    }
+
+    // Outbound connections should be prioritized. This prioritizes
+    // other VFNs/seed peers over regular PFNs. Inbound connections
+    // are always low priority (as they are generally unreliable).
+    if let Some(metadata) = utils::get_metadata_for_peer(&peers_and_metadata, *peer) {
+        if metadata.get_connection_metadata().is_outbound_connection() {
+            PeerPriority::HighPriority
+        } else {
+            PeerPriority::LowPriority
+        }
+    } else {
+        PeerPriority::LowPriority // We don't have connection metadata
+    }
+}
+
+/// Returns true iff the given peer is a trusted peer
+fn is_trusted_peer(peers_and_metadata: Arc<PeersAndMetadata>, peer: &PeerNetworkId) -> bool {
+    peers_and_metadata
+        .get_trusted_peer_state(peer)
+        .is_ok_and(|peer_state| peer_state.is_some())
+}
+
+/// Returns true iff the specified peer is a high priority peer.
+/// This is a useful utility function.
```

Comment:
nit: remove?

Reply:
Aah, thanks -- this was a legacy comment that no longer makes any sense! 😄